### PR TITLE
Fixes scaling NSE

### PR DESCRIPTION
**Pull request by jimrojerTNO**
- Bug in nominals for current
- bugs in electricity cable nominals
- fix in gas pipe nominals
- fix for setting nominals for node when nodes have logical links

### CODE REVIEW

File: src/mesido/pycml/component_library/milp/electricity/electricity_cable.py [hunk 1]
```diff
@@ -31,8 +31,13 @@ def __init__(self, name, **modifiers):
         self.nominal_current = nan
         self.nominal_voltage = nan
         self.r = 1.0e-6 * self.length  # TODO: temporary value
-        self.nominal_voltage_loss = (self.nominal_current * self.r * self.nominal_current) ** 0.5
-        self.add_variable(Variable, "Power_loss", min=0.0, nominal=self.r * self.max_current**2)
+        self.nominal_voltage_loss = (self.nominal_current * self.r * self.nominal_voltage) ** 0.5
+        self.loss_nominal = self.r * self.max_current * self.nominal_current
```
Comment (FJanssen-TNO) [inline src/mesido/pycml/component_library/milp/electricity/electricity_cable.py:35]: From the variable name it is now unclear what on which variable this loss is occurring, maybe change it to power_loss_nominal. And furthermore consider whether this parameter needs to be saved separately as it is also set the nominal of the power_loss

File: src/mesido/esdl/asset_to_component_base.py [hunk 2]
```diff
@@ -510,23 +510,43 @@ def _get_connected_i_nominal_and_max(self, asset: Asset) -> Tuple[float, float]:
...
             i_nom_out = (
                 self._port_to_i_nominal.get(connected_port, None)
                 if self._port_to_i_nominal.get(connected_port, False)
-                else max([asset.attributes.get(key, -1) for key in self.__power_keys])
+                else max(
+                    [
+                        asset.attributes.get(key, -1) / connected_port.carrier.voltage
+                        for key in self.__power_keys
+                    ]
+                )
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply src/mesido/esdl/asset_to_component_base.py:528]: agree

File: src/mesido/pycml/component_library/milp/electricity/electricity_cable.py (hunk 1 above)
Comment (jimrojerTNO) [reply src/mesido/pycml/component_library/milp/electricity/electricity_cable.py:35]: changed the name, as we use this variable a few lines lower to check scaling ranges I suggest we keep it saved here

File: src/mesido/electricity_physics_mixin.py [hunk 3]
```diff
@@ -166,6 +169,37 @@ def pre(self):
...
+                            ] = nominals[var][i]
+                        else:
+                            # Here we set a nominal based on median of all the connected assets to
+                            # the node. This is specifically done when we have a logical link for
+                            # node to node. In this case we cannot set the nominal based on the
+                            # connected node, hence we assume a node has at least one not node
+                            # asset connected to it.
+                            self.__bus_variable_nominal[
+                                f"{node}.ElectricityConn[{i + 1}].{var}"
+                            ] = np.median(nominals[var])
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply src/mesido/electricity_physics_mixin.py:201]: agree

File: src/mesido/gas_physics_mixin.py [hunk 4]
```diff
@@ -271,6 +275,37 @@ def _get_min_bound(bound):
...
+                            )
+                        else:
+                            # Here we set a nominal based on median of all the connected assets to
+                            # the node. This is specifically done when we have a logical link for
+                            # node to node. In this case we cannot set the nominal based on the
+                            # connected node, hence we assume a node has at least one not node
+                            # asset connected to it.
+                            self.__gas_node_variable_nominal[f"{node}.GasConn[{i + 1}].{var}"] = (
+                                np.median(nominals[var])
+                            )
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply src/mesido/gas_physics_mixin.py:307]: agree

File: src/mesido/heat_physics_mixin.py [hunk 5]
```diff
@@ -545,6 +548,37 @@ def _get_min_bound(bound):
...
+                        else:
+                            # Here we set a nominal based on median of all the connected assets to
+                            # the node. This is specifically done when we have a logical link for
+                            # node to node. In this case we cannot set the nominal based on the
+                            # connected node, hence we assume a node has at least one not node
+                            # asset connected to it.
+                            self.__heat_node_variable_nominal[f"{node}.HeatConn[{i + 1}].{var}"] = (
+                                np.median(nominals[var])
+                            )
+
```
[the remark this answers is not in the document]
Comment (jimrojerTNO) [reply src/mesido/heat_physics_mixin.py:581]: agree

File: src/mesido/esdl/asset_to_component_base.py [hunk 6]
```diff
@@ -545,12 +565,22 @@ def _get_connected_i_nominal_and_max(self, asset: Asset) -> Tuple[float, float]:
...
             )
             i_nom = (
                 self._port_to_i_nominal.get(connected_port, None)
                 if self._port_to_i_nominal.get(connected_port, False)
-                else max([asset.attributes.get(key, -1) for key in self.__power_keys])
+                else max(
+                    [
+                        asset.attributes.get(key, -1) / connected_port.carrier.voltage
+                        for key in self.__power_keys
+                    ]
```
Comment (FJanssen-TNO) [inline src/mesido/esdl/asset_to_component_base.py:582]: This one should also have a divide by 2.0

File: src/mesido/esdl/asset_to_component_base.py [hunk 7]
```diff
@@ -572,12 +602,22 @@ def _get_connected_i_nominal_and_max(self, asset: Asset) -> Tuple[float, float]:
...
                     )
                     i_nom = (
                         self._port_to_i_nominal.get(connected_port, None)
                         if self._port_to_i_nominal.get(connected_port, False)
-                        else max([asset.attributes.get(key, -1) for key in self.__power_keys])
+                        else max(
+                            [
+                                asset.attributes.get(key, -1) / connected_port.carrier.voltage
+                                for key in self.__power_keys
+                            ]
```
Comment (FJanssen-TNO) [inline src/mesido/esdl/asset_to_component_base.py:619]: divide by 2.0

Comment (FJanssen-TNO) [reply src/mesido/esdl/asset_to_component_base.py]: updated with agreement from @jimrojerTNO

File: src/mesido/esdl/asset_to_component_base.py (hunk 6 above)
Comment (FJanssen-TNO) [reply src/mesido/esdl/asset_to_component_base.py]: updated with agreement from @jimrojerTNO

**Review by FJanssen-TNO — APPROVED** [review]
